indexer/src/main/java/au/org/aodn/esindexer/service/AodnDiscoveryParameterVocabService.java

```diff
@@ -0,0 +1,70 @@
+package au.org.aodn.esindexer.service;
+
+import au.org.aodn.esindexer.abstracts.OgcApiRequestEntityCreator;
+import au.org.aodn.esindexer.utils.CacheArdcVocabsUtils;
+import au.org.aodn.stac.model.ConceptModel;
+import au.org.aodn.stac.model.ThemesModel;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class AodnDiscoveryParameterVocabService {
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    OgcApiRequestEntityCreator ogcApiRequestEntityCreator;
+
+    @Autowired
+    CacheArdcVocabsUtils cacheArdcVocabsUtils;
+
+    protected boolean themesMatchConcept(List<ThemesModel> themes, ConceptModel concept) {
+        for (ThemesModel theme : themes) {
+            for (ConceptModel themeConcept : theme.getConcepts()) {
+                /*
+                comparing by hashcode (id and url) of the concept object
+                this will prevent cases where bottom-level vocabs are the same in text, but their parent vocabs are different
+                e.g "car -> parts" vs "bike -> parts" ("parts" is the same but different parent)
+                 */
+                if (themeConcept.equals(concept)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public List<String> getAodnDiscoveryCategories(List<ThemesModel> themes) throws IOException {
```

Comment:
Can add comment on what it is trying to do? Although I know it is trying to get the second level but other may not easily know

Reply:
surething